nemo/collections/nlp/models/intent_slot_classification/intent_slot_classification_model.py

```diff
@@ -53,70 +53,124 @@ def output_types(self) -> Optional[Dict[str, NeuralType]]:
     def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         """ Initializes BERT Joint Intent and Slot model.
         """
-
-        self.data_dir = cfg.data_dir
         self.max_seq_length = cfg.language_model.max_seq_length
 
-        self.data_desc = IntentSlotDataDesc(
-            data_dir=cfg.data_dir, modes=[cfg.train_ds.prefix, cfg.validation_ds.prefix]
-        )
+        # Check the presence of data_dir.
+        if not cfg.data_dir:
```

Comment:
Maybe also add `or not os.path.exists(cfg.data_dir)` ? For the models trained on ngc and then restored, the config will have data_dir with an invalid path

Reply:
👍

Reply:
Still, the "restore path" is handled in a totally different way...

Reply:
But is it working? during restoration `cfg.data_dir` won't be None and `_copy_data_desc_to_cfg` will be attempted, no?

Reply:
I have tested it in TLT in: training (with two ways injecting data_dir), finetuning, evaluation and inference. Plus it is passing CI in NeMo. So I would say it is working... but still need to be refactored ;)